cpp/arcticdb/column_store/column.cpp

```diff
@@ -366,23 +366,25 @@ void Column::mark_absent_rows(size_t num_rows) {
 }
 
 void Column::default_initialize_rows(size_t start_pos, size_t num_rows, bool ensure_alloc) {
-    type_.visit_tag([that=this, start_pos, num_rows, ensure_alloc](auto tag) {
-        using T= std::decay_t<decltype(tag)>;
-        using RawType = typename T::DataTypeTag::raw_type;
-        const auto bytes = (num_rows * sizeof(RawType));
+    if (num_rows > 0) {
+        type_.visit_tag([that = this, start_pos, num_rows, ensure_alloc](auto tag) {
+            using T = std::decay_t<decltype(tag)>;
+            using RawType = typename T::DataTypeTag::raw_type;
+            const auto bytes = (num_rows * sizeof(RawType));
 
-        if(ensure_alloc)
-            that->data_.ensure<uint8_t>(bytes);
+            if (ensure_alloc)
+                that->data_.ensure<uint8_t>(bytes);
 
```

Comment:
Could get rid of that->, but as you didn't do it we can always catch it on a subsequent PR. Maybe fix it if you have to re-push anyway